File: software/examples/perl/example_simple.pl
#!/usr/bin/perl

use strict;
use Tinkerforge::IPConnection;
use Tinkerforge::BrickletSoundPressureLevel;

use constant HOST => 'localhost';
use constant PORT => 4223;
use constant UID => 'XYZ'; # Change XYZ to the UID of your Sound Pressure Level Bricklet

my $ipcon = Tinkerforge::IPConnection->new(); # Create IP connection
my $spl = Tinkerforge::BrickletSoundPressureLevel->new(&UID, $ipcon); # Create device object

$ipcon->connect(&HOST, &PORT); # Connect to brickd
# Don't use device before ipcon is connected

# Get current decibel
my $decibel = $spl->get_decibel();
print "Decibel: " . $decibel/10.0 . " dB(A)\n";

print "Press key to exit\n";
<STDIN>;
$ipcon->disconnect();

File: software/examples/ruby/example_simple.rb
#!/usr/bin/env ruby
# -*- ruby encoding: utf-8 -*-

require 'tinkerforge/ip_connection'
require 'tinkerforge/bricklet_sound_pressure_level'

include Tinkerforge

HOST = 'localhost'
PORT = 4223
UID = 'XYZ' # Change XYZ to the UID of your Sound Pressure Level Bricklet

ipcon = IPConnection.new # Create IP connection
spl = BrickletSoundPressureLevel.new UID, ipcon # Create device object

ipcon.connect HOST, PORT # Connect to brickd
# Don't use device before ipcon is connected

# Get current decibel
decibel = spl.get_decibel
puts "Decibel: #{decibel/10.0} dB(A)"

puts 'Press key to exit'
$stdin.gets
ipcon.disconnect
